CONTRIBUTING.md

```diff
@@ -1,16 +1,518 @@
+
+    ```
+    git clone git@github:cyberark/conjur-google-cloud-marketplace
+    ```
+
+2. Update the Google Cloud Launcher [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules):
+
+    ```
+    git submodule sync --recursive
+    git submodule update --recursive --init --force
+    ```
+
+3. Create a local development branch
+
+    For example:
+
+    ```
+    git checkout -b my-cool-fix
+    ```
+
+4. Make local changes to your fork by editing files
+
+5. Build and test the component Docker images as described below in [Building and Testing](#building-and-testing).
+
+6. [Commit your changes](https://help.github.com/en/github/managing-files-in-a-repository/adding-a-file-to-a-repository-using-the-command-line)
+
+    For example:
+
+    ```
+    git commit -a -m "Adds a cool fix"
+    ```
+
+7. [Push your local changes to the remote repository](https://help.github.com/en/github/using-git/pushing-commits-to-a-remote-repository)
+
+    For example:
+
+    ```
+    git push origin my-cool-fix
+    ```
+
+8. [Create new Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)
 
 From here your pull request will be reviewed and once you've responded to all
 feedback it will be merged into the project. Congratulations, you're a
 contributor!
+
+## Building, Deploying, and Testing
+
+Once you have completed Steps 1-4 in the [Pull Request Workflow](#pull-request-workflow) section above, you are ready to build the Conjur Google Marketplace Application, and test it on a Kubernetes platform.
+
+### Create Namespace for Testing (Optional)
+
+If you would like the namespace in which the application will run to have
+a convenient name, create a namespace e.g. as follows:
+
+```
+export NAMESPACE="$(whoami)"
+kubectl create ns "$NAMESPACE"
+kubectl config set-context --current --namespace="$NAMESPACE"
+```
+If you choose not to create a namespace as shown above, a namespace with
+a random name (of the form "apptest-XXXXXX") will be created.
+
+### Configure Docker Registry
+
+If you are using your own registry, run the following:
+
+```
+cd conjur-google-cloud-marketplace  # Root dir of this repository
+export REGISTRY=<your-docker-registry>
+export FLAT_REGISTRY=true
+```
+The setting of `FLAT_REGISTRY=true` is required because Dockerhub
+registries typically require images to be stored in a flat
+registry namespace (as opposed to Google Cloud Registry,
+where images are stored in project directories).
+
+Otherwise, if you are using the Google Cloud Registry, run:
+
+```
+cd conjur-google-cloud-marketplace
+export REGISTRY=gcr.io/conjur-cloud-launcher-onboard
+```
+
+### Build, Deploy, and Test the Application
+
+#### Build, Deploy, and Leave the Application Running
+
+Follow these steps if you would like to build and deploy the application
+and leave it running (for manual verification).
+
+1. Build and deploy the application with the `--persist` flag:
+
+   ```
```

Comment:
Some of the new code snippets need this prefix to hint to the viewer how to style it:
```
```sh-session
```